src/main/scala/ch/epfl/k2sjsir/translate/GenBinary.scala

```diff
@@ -173,7 +173,9 @@ object GenBinary {
     "EQEQ" -> BinaryOp.Boolean_==,
     "EXCLEQ" -> BinaryOp.Boolean_!=,
     "or" -> BinaryOp.Boolean_|,
-    "and" -> BinaryOp.Boolean_&
+    "and" -> BinaryOp.Boolean_&,
+    "ANDAND" -> BinaryOp.Boolean_&,
+    "OROR" -> BinaryOp.Boolean_|
```

Comment:
I think this is not correct. Assuming ` ANDAND` and `OROR` are supposed to be *short-circuiting*, you cannot compile them like that. They must be compiled as `if (lhs) rhs else false` and `if (lhs) true else rhs`, respectively.